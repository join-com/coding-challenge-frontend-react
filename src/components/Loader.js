import React from 'react';
import Status from './Status';

function Loader(Component) {
	/*eslint-disable react/display-name */
	return class extends React.Component {
		static defaultProps = {
			code: Status.LOADING,
		}
		render() {
			let { code, ...props } = this.props;
			if (code === Status.ERROR) {
				return <p>Opps, an error occurred.</p>;
			} else if (code === Status.FAILURE) {
				return <p>No data found.</p>;
			} else if (code === Status.SUCCESS) {
				return <Component {...props} />;
			} else {
				return <p>Fetching incident(s). Please wait...</p>;
			}

		}
	}
}

export default Loader;