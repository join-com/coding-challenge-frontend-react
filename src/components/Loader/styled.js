import styled from 'styled-components'

const Container = styled.div`
  margin: 20px 0;
`

export { Container }
