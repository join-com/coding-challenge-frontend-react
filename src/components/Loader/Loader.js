import React from 'react'
import { Spin } from 'antd'

import { Container } from './styled'

const Loader = () => (
  <Container>
    <Spin tip='Loading...' />
  </Container>
)

export default Loader
