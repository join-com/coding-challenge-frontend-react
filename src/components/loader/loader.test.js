import React from 'react';
import { shallow , mount} from 'enzyme';
import Loader from './loader';
import CircularProgress from '@material-ui/core/CircularProgress';

describe('Loader', () => {
    it('it should render correctly', () => {
        const component = shallow(<Loader />);
        expect(component).toMatchSnapshot();
    });
    
    it('props are correct ', () => {
        let testSize = 30;
        const component = mount(<Loader size={testSize}/>);
        expect(component.props().size).toEqual(testSize);
    });

    it('renders `in-place-loader',()=>{
        const component = shallow(<Loader />);
        expect(component.find('.in-place-loader').length).toBe(1)
    })

    it('renders `<CircularProgress />',()=>{
        const component = shallow(<Loader />);
        expect(component.find(CircularProgress).length).toBe(1)
    })
    
});