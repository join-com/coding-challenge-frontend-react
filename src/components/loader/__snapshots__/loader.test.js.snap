// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader it should render correctly 1`] = `
<div
  className="in-place-loader"
>
  <WithStyles(CircularProgress)
    thickness={5}
  />
</div>
`;
